contracts/staking/StakingPool.sol

```diff
@@ -0,0 +1,594 @@
+ * while fixed tokens rewards are calculated once per user.
+ */
+contract StakingPool is
+    PausableUpgradeable,
+    ReentrancyGuard,
+    RoleAccessControl
+{
+    // Magic Number fixed length rewardsAmounts to fit 3 words. Only used here.
+    struct User {
+        uint128 depositAmount;
+        uint128[5] rewardAmounts;
+    }
+
+    struct RewardOwed {
+        address tokens;
+        uint128 amount;
+    }
+
+    mapping(address => User) private _users;
+
+    uint32 private _rewardsAvailableTimestamp;
+    bool private _emergencyMode;
+    uint128 private _totalStakedAmount;
+
+    StakingPoolLib.Config private _stakingPoolConfig;
+
+    event WithdrawRewards(
+        address indexed user,
+        address rewardToken,
+        uint256 rewards
+    );
+    event WithdrawStake(address indexed user, uint256 stake);
+    event Deposit(address indexed user, uint256 depositAmount);
+    event InitializeRewards(address rewardTokens, uint256 amount);
+    event RewardsAvailableTimestamp(uint32 rewardsAvailableTimestamp);
+    event EmergencyMode(address indexed admin);
+    event NoRewards(address indexed user);
+
+    modifier rewardsAvailable() {
+        require(_isRewardsAvailable(), "StakingPool: rewards too early");
+        _;
+    }
+
+    modifier stakingPeriodComplete() {
+        require(_isStakingPeriodComplete(), "StakingPool: still stake period");
+        _;
+    }
+
+    modifier stakingPoolRequirementsUnmet() {
+        //slither-disable-next-line timestamp
+        require(
+            (_totalStakedAmount < _stakingPoolConfig.minTotalPoolStake) &&
+                (block.timestamp > _stakingPoolConfig.epochStartTimestamp),
+            "StakingPool: requirements unmet"
+        );
+        _;
+    }
+
+    modifier emergencyModeEnabled() {
+        require(_emergencyMode, "StakingPool: not emergency mode");
+        _;
+    }
+
+    function pause()
+        external
+        whenNotPaused
+        atLeastDaoMeepleRole(_stakingPoolConfig.daoId)
+    {
+        _pause();
+    }
+
+    function unpause()
+        external
+        whenPaused
+        atLeastDaoMeepleRole(_stakingPoolConfig.daoId)
+    {
+        _unpause();
+    }
+
+    /**
+     * @notice Only entry point for a user to deposit into the staking pool
+     *
+     * @param amount Amount of stake tokens to deposit
+     */
+    function deposit(uint256 amount) external whenNotPaused nonReentrant {
+        StakingPoolLib.Config storage _config = _stakingPoolConfig;
+
+        require(
+            amount >= _config.minimumContribution,
+            "StakingPool: min contribution"
+        );
+        require(
+            _totalStakedAmount + amount < _config.maxTotalPoolStake,
+            "StakingPool: oversubscribed"
+        );
+        //slither-disable-next-line timestamp
+        require(
+            block.timestamp < _config.epochStartTimestamp,
```

Comment:
The intention of this require statement is to prevent depositing when the staking period has started, as per the spec. This also prevents depositing after the staking period has finished as this timestamp will also be greater than epochStartTimestamp.

Reply:
I see, I misunderstood the purpose of the check that makes sense.